src/main/scala/com/gu/googleamp/AmpFlusher.scala

```diff
@@ -1,25 +1,35 @@
 package com.gu.googleamp
 
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.util.IOUtils
 import okhttp3.{ OkHttpClient, Request }
 import org.joda.time.DateTime
+
 import java.security.spec.PKCS8EncodedKeySpec
 import java.security.{ KeyFactory, PrivateKey, Signature }
 import com.gu.fastly.Config
+import com.gu.fastly.Config.s3
 
 object AmpFlusher {
 
   // This object implements AMP flush/delete according to https://developers.google.com/amp/cache/update-cache
 
+  private val s3 = AmazonS3ClientBuilder.defaultClient
   private val httpClient = new OkHttpClient()
-  private val config = Config.load()
 
   def getCurrentUnixtime(): Long = {
     DateTime.now().getMillis() / 1000
   }
 
+  private def readPrivateKeyFromS3(): Array[Byte] = {
+    val inputStream = s3.getObject("fastly-cache-purger-config", "amp-flusher-private-key.der").getObjectContent()
```

Comment:
I disagree with AWS native code like this been buried in application code. The Config object is hiding the fact that the config is in S3.

AMPFlusher 'doesn't need' S3 to run; it needs a private key.
An old school Java way to express this would be for AMPFlusher to express all of it's requirements on it's contructor; like:
AmpFlusher(privateKey: String).

Then the user, the Fastly decacher could construct it using whatever config system it happened to use.

ie. amp = AmpFlusher(config.ampPrivateKey)

AmpFlusher doesn't really need to be tried to Amazon S3 just because Fastly happens to be at this point in time.

Reply:
Fair enough :)